### PR TITLE
First attempt at global randomization

### PR DESCRIPTION
**Pull request by Oipo**

### CODE REVIEW

File: rando/src/main.rs [hunk 1]
```diff
@@ -101,13 +136,20 @@ fn main() -> Result<(), Error> {
...
+    chest_pairs.shuffle(&mut rng);
+
+    for i in 0..=0xf {
+        write_new_chests_for_area(&mut rng, &n, i, &mut buffer, &mut chest_pairs)?;
     }
 
     let filename = &opt
         .out
-        .unwrap_or_else(|| PathBuf::from(format!("NR-{:#}.pce", radix_36(seed))));
+        .unwrap_or_else(|| PathBuf::from(format!("neutopia-NR-{:#}.pce", radix_36(seed))));
```
Comment (konkers) [inline rando/src/main.rs:152]: My thinking here was NR stood for Neutopia Randomizer.

File: rando/src/main.rs [hunk 2]
```diff
@@ -45,19 +78,21 @@ fn shuffle_area(
...
                     }
                 }
             }
         }
     }
 
-    // Next shuffle the chest contents.
     chest_contents.shuffle(rng);
 
-    // Patch up a new chest table.
```
Comment (konkers) [inline rando/src/main.rs:60]: Let's keep this comment.

File: rando/src/main.rs [hunk 3]
```diff
@@ -101,13 +136,20 @@ fn main() -> Result<(), Error> {
 
     let n = Neutopia::new(&buffer)?;
 
-    for i in 4..=0xb {
-        shuffle_area(&mut rng, &n, i, &mut buffer)?;
+    let mut chest_pairs = Vec::new();
```
Comment (konkers) [inline rando/src/main.rs:139]: Why the name `chest_pairs`?  Maybe a previous version of `get_randomizable_chests_for_area` return a tuple?

File: rando/src/main.rs [hunk 4]
```diff
@@ -23,16 +23,49 @@ struct Opt {
     seed: Option<String>,
 }
 
-fn shuffle_area(
+/*
+   Returns a list of chests that are allowed to be randomized (so no crypt keys, crystal balls or medallions are returned here)
+   All these chests are piled on one big heap, randomized and passed into write_new_chests_for_area.
+*/
```
Comment (konkers) [inline rando/src/main.rs:29]: I'm pretty sure it is idomatic in Rust to prefer line comments of block comments.  I've been sticking to that elsewhere in the project.

File: rando/src/main.rs [hunk 5]
```diff
@@ -45,19 +78,21 @@ fn shuffle_area(
...
+                            return Err(format_err!("Terrible error, ran out of chests! T_T"));
+                        }
                     }
                 }
             }
         }
     }
 
-    // Next shuffle the chest contents.
     chest_contents.shuffle(rng);
```
Comment (konkers) [inline rando/src/main.rs:94]: I think this shuffle is unneeded because `randomizable_chests` is shuffled in main.

**Review by konkers — CHANGES_REQUESTED** [review]
Good approach for a first pass at global item rando!

File: rando/src/main.rs (hunk 1 above)
Comment (Oipo) [reply rando/src/main.rs:152]: Yeah, sveniroth and I got confused with the name.

File: rando/src/main.rs (hunk 3 above)
Comment (Oipo) [reply rando/src/main.rs:139]: You got that correct, lol. I'll rename it.

File: rando/src/main.rs (hunk 5 above)
Comment (Oipo) [reply rando/src/main.rs:94]: Yeah, it should've been the new chest table

**Pull request closed without merging**